concent_api/api_testing_common.py

```diff
@@ -185,21 +206,24 @@ def _print_data(data: Optional[Union[bytes, str]], url: str) -> None:
             print('SEND ({})'.format(url))
             print('MESSAGE:')
             print_golem_message(data)
-
+    if headers is None:
+        headers = REQUEST_HEADERS
     assert all(value not in ['', None] for value in [endpoint, host, REQUEST_HEADERS])
     url = "{}/api/v1/{}/".format(host, endpoint)
-
     _print_data(data, url)
-    response = requests.post("{}".format(url), headers=REQUEST_HEADERS, data=_prepare_data(data), verify=False)
+    response = requests.post("{}".format(url), headers=headers, data=_prepare_data(data), verify=False)
     _print_response(private_key, public_key, response)
     validate_response_status(response.status_code, expected_status)
     validate_content_type(response.headers['Content-Type'], expected_content_type)
     validate_response_message(response.content, expected_message_type, private_key, public_key)
+    if 'Concent-Golem-Messages-Version' in response.headers:
```

Comment:
Maybe we should we print something when this header is missing?
And I see now below that `expected_golem_version` can be None with this flow. I think it would be better to add to this `if `:
```
if expected_golem_version is not None and 'Concent-Golem-Messages-Version' in response.headers:
```
and the make it non-optional in function.

Reply:
This header is missing when we get http404 responses for example, so what do You suggest we should print about that?

Reply:
You can now remove `if expected_golem_version is not None:` from the `validate_golem_version`.

Reply:
Right! Done.